docs/reference/mapping/types/alias.asciidoc

```diff
@@ -0,0 +1,105 @@
+[[alias]]
+=== Alias datatype
+
+NOTE: Field aliases can only be specified on indexes with a single mapping type. To add a field
+alias, the index must therefore have been created in 6.0.0 or later, or be an older index with
+the setting `index.mapping.single_type: true`. Please see <<removal-of-types>> for more information.
```

Comment:
I wasn't sure whether we should accept this setting, or if we should strictly enforce that the index be created in 6.x or later. I'm concerned that there will be confusing behavior in the following case: this setting is enabled, a field alias is added, then the setting is disabled.